### PR TITLE
task/create tablayout bill pending

### PR DESCRIPTION
**Pull request by hungvv-1301**
## Related Tickets
- [#Task 54683: Create item recycler view bill](https://edu-redmine.sun-asterisk.vn/issues/54683)
- [#Task 54684: Screen pending confirm with logic](https://edu-redmine.sun-asterisk.vn/issues/54684)
## WHAT
- Create tablayout bill screen
- Create bill pending screen
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/atom/android/bookshop/data/repository/BillRepository.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        token: String?,
+        idBill: Int,
+        status: Int,
+        callback: IRequestCallback<ResponseObject<Bill>>
+    ) {
+        remote.updateStatusBill(token, idBill, status, callback)
+    }
+
+    companion object {
+        private var instance: BillRepository? = null
```
Comment (ghost) [inline app/src/main/java/com/atom/android/bookshop/data/repository/BillRepository.kt:27]: blank line

File: app/src/main/java/com/atom/android/bookshop/data/source/IBillDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.atom.android.bookshop.data.source
+
+import com.atom.android.bookshop.data.model.Bill
+import com.atom.android.bookshop.data.source.remote.IRequestCallback
+import com.atom.android.bookshop.data.source.remote.ResponseObject
+
+interface IBillDataSource {
+    interface Remote {
```
Comment (ghost) [inline app/src/main/java/com/atom/android/bookshop/data/source/IBillDataSource.kt:7]: .

File: app/src/main/java/com/atom/android/bookshop/data/source/remote/api/ApiURL.kt [hunk 3]
```diff
@@ -6,7 +6,10 @@ object ApiURL {
 
     fun pathLogin(): String = BuildConfig.API_HOST + ApiConstants.Endpoint.LOGIN
     fun pathForgotEmail(): String = BuildConfig.API_HOST + ApiConstants.Endpoint.FORGOT_EMAIL
+    fun pathOrder(): String = BuildConfig.API_HOST + ApiConstants.Endpoint.ORDER
```
Comment (ghost) [inline app/src/main/java/com/atom/android/bookshop/data/source/remote/api/ApiURL.kt:9]: .

File: app/src/main/java/com/atom/android/bookshop/data/source/remote/bill/BillRemoteDataSource.kt [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        } catch (ex: JSONException) {
+            throw JSONConvertException(ex)
+        } catch (ex: IllegalStateException) {
+            throw JSONConvertException(ex)
+        }
+    }
+
+    companion object {
+        private var instance: BillRemoteDataSource? = null
+        fun getInstance() = synchronized(this) {
```
Comment (ghost) [inline app/src/main/java/com/atom/android/bookshop/data/source/remote/bill/BillRemoteDataSource.kt:117]: .

File: app/src/main/java/com/atom/android/bookshop/ui/bill/pending/BillPendingPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+                override fun onFailed(message: String?) {
+                    view.destroyBillFailed(message)
+                }
+            })
+    }
+
+
+    companion object {
+        private var instance: BillPendingPresenter? = null
```
Comment (ghost) [inline app/src/main/java/com/atom/android/bookshop/ui/bill/pending/BillPendingPresenter.kt:70]: .

File: app/src/main/java/com/atom/android/bookshop/data/model/Bill.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    val orderLines: List<OrderLine>,
+    val createdAt: String,
+) {
+    fun totalBill(): Double {
+        var totalBill = 0.0
+        orderLines.forEach { orderLine ->
+            totalBill += orderLine.amount * orderLine.price
+        }
+        return totalBill
+    }
```
Comment (minhnn-1279) [inline app/src/main/java/com/atom/android/bookshop/data/model/Bill.kt]: fun totalBill(): Double = orderLines.sumOf { it.amount * it.price }

nếu trả về Int thì toDouble :v

File: app/src/main/java/com/atom/android/bookshop/data/model/Bill.kt [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    fun totalItem(): Int {
+        var totalItem = 0
+        orderLines.forEach { orderLine ->
+            totalItem += orderLine.amount
+        }
+        return totalItem
+    }
```
Comment (minhnn-1279) [inline app/src/main/java/com/atom/android/bookshop/data/model/Bill.kt:28]: xử lý tương tự bên trên nhé

File: app/src/main/res/layout/fragment_bill.xml [hunk 8]
```diff
@@ -3,12 +3,51 @@
...
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:text="@string/title_bill" />
+        android:layout_height="match_parent">
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:id="@+id/layout_toolbar"
+            android:layout_width="match_parent"
+            android:layout_height="?actionBarSize"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_bill.xml:18]: fomat code

File: app/src/main/java/com/atom/android/bookshop/data/source/remote/bill/BillRemoteDataSource.kt [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+            } else {
+                handler.post {
+                    callback.onFailed(responseObject.message)
+                }
+            }
+        } catch (ex: JSONException) {
+            throw JSONConvertException(ex)
+        } catch (ex: IllegalStateException) {
+            throw JSONConvertException(ex)
+        }
```
Comment (minhnn-1279) [inline app/src/main/java/com/atom/android/bookshop/data/source/remote/bill/BillRemoteDataSource.kt:112]: t nghĩ nếu muốn try/catch thì ông bắt try/catch lớp CallAPI nó gọn hơn

File: app/src/main/java/com/atom/android/bookshop/ui/bill/BillFragment.kt [hunk 10]
```diff
@@ -1,27 +1,33 @@
...
-    override fun initialize() {
-        // TODO implement later
-    }
 
     override fun initView() {
-        // TODO implement later
+        binding?.viewPagerBill?.adapter = ViewPagerTabLayout(
+            childFragmentManager,
+            FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT,
+            listOf<Fragment>(
```
Comment (minhnn-1279) [inline app/src/main/java/com/atom/android/bookshop/ui/bill/BillFragment.kt]:
```suggestion
            listOf(
```

File: app/src/main/java/com/atom/android/bookshop/ui/bill/pending/ListAdapterBillPending.kt [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+import com.atom.android.bookshop.R
+import com.atom.android.bookshop.base.BaseAdapter
+import com.atom.android.bookshop.base.BaseViewHolder
+import com.atom.android.bookshop.data.model.Bill
+import com.atom.android.bookshop.databinding.ItemBillBinding
+import com.atom.android.bookshop.utils.Constants
+import com.atom.android.bookshop.utils.convertStrToMoney
+
+class ListAdapterBillPending(
+    private val onClick: (Bill) -> Unit,
```
Comment (minhnn-1279) [inline app/src/main/java/com/atom/android/bookshop/ui/bill/pending/ListAdapterBillPending.kt:15]: Để ra typeAlias cho gọn á :v

**Review by ndt0409 — APPROVED** [review]

**Pull request merged**